### PR TITLE
Add a CC-by-4.0 License.

### PR DESCRIPTION
**Pull request by samcunliffe**

### CODE REVIEW

**Review by stefpiatek — APPROVED** [review]
CC is very long isn't it. Looks good

File: README.md [hunk 1]
```diff
@@ -5,3 +5,7 @@ These are our default assumed style choices unless there are existing project- o
 
   * [Python](python)
   * [R](r)
+
+---
+
+<p xmlns:cc="http://creativecommons.org/ns#" xmlns:dct="http://purl.org/dc/terms/"><span property="dct:title">The UCL-ARC coding standards guidelines</span> by <span property="cc:attributionName">the Centre for Advanced Research Computing, UCL</span> are licensed under <a href="http://creativecommons.org/licenses/by/4.0/?ref=chooser-v1" target="_blank" rel="license noopener noreferrer" style="display:inline-block;">CC BY 4.0<img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1"><img style="height:22px!important;margin-left:3px;vertical-align:text-bottom;" src="https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1"></a></p>
```
Comment (matt-graham) [inline README.md:11]: It looks like GitHub at least strips a lot of the attributes out from this in rendered Markdown - not sure if Jekyll does similarly or not. Would possibly suggest to simplifying to something like
```suggestion
The UCL-ARC coding standards guidelines by the Centre for Advanced Research Computing, UCL are licensed under [CC BY 4.0](http://creativecommons.org/licenses/by/4.0/?ref=chooser-v1)<img style="height:22px;" src="https://mirrors.creativecommons.org/presskit/icons/cc.svg?ref=chooser-v1"><img style="height:22px;" src="https://mirrors.creativecommons.org/presskit/icons/by.svg?ref=chooser-v1">
```
to keep it also readable in Markdown source? But not important

**Review by matt-graham — APPROVED** [review]
All fine by me. To complicate things further, I did a _bad thing_ 😳 by copying a snippet from our internal documentation ([the original text of the Python recommendations](https://github.com/UCL-ARC/research-software-documentation/blob/186b74573b22a43fd7ee4b4664bfd56fe333c1ed/processes/programming_projects/group_practices.md?plain=1#L56-L58)) without checking in with the original authors first if they were happy for this to be released publicly. From looking through [the blame for these lines](https://github.com/UCL-ARC/research-software-documentation/blame/186b74573b22a43fd7ee4b4664bfd56fe333c1ed/processes/programming_projects/group_practices.md#L56-L58) it looks like (in addition to @samcunliffe) the people who contributed to these lines were @dstansby, @jonc125 and @jamespjh (👋) so to make sure we dot all our i's we should also get their explicit consent for releasing under this licence.

**Comment by matt-graham** [comment]
Tagged relevant people for reviews under assumption that clicking approve `==` giving their consent.

**Review by dstansby — APPROVED** [review]

**Review by milanmlft — APPROVED** [review]

**Review by jonc125 — APPROVED** [review]
I'm happy with the licence choice.

**Comment by jamespjh** [comment]
So I think we should probably put a license on the ARC internal doc, and open source as much of it as we can, to facilitate the wider RSE community. Hardly any of it needs to be secret, and ARC's ways of working should be a useful open template on how to be an RSE organisation.

Anyway, of course approved.

**Review by jamespjh — COMMENTED** [review]
Approved

**Pull request merged**